cassandra/cluster.py

```diff
@@ -3679,6 +3718,11 @@ def _try_connect(self, host):
             shared_results = (peers_result, local_result)
             self._refresh_node_list_and_token_map(connection, preloaded_results=shared_results)
             self._refresh_schema(connection, preloaded_results=shared_results, schema_agreement_wait=-1)
+
+            if self._cluster.check_tablets_enabled() and self._cluster.check_tablets_support(connection, self._timeout):
+                tablets_query = QueryMessage(query=self._SELECT_TABLETS, consistency_level=ConsistencyLevel.ONE)
+                tablets_result = connection.wait_for_response(tablets_query, timeout=self._timeout)
+                self._refresh_tablets(connection, preloaded_result=tablets_result)
```

Comment:
What happens if `check_tablets_support` succeeds once and fails one? We get a `Cluster` that is in a very weird state.
I think we can fix this and avoid having `check_tablets_support` function at all.
It would look like this:
```python
if self._cluster.check_tablets_enabled():
    try:
        self._refresh_tablets(connection, preloaded_result=tablets_result)
    except table_does_not_exist: # idk what the name is, you need to check
        # Tablet table does not exist - cluster doesn't support tablets
        self._cluster.experimental_tablet_feature_enabled
   # Other exceptions should not be caught - if there was a timeout while trying to fetch tablets, `Cluster` initialization should fail.
    self._cluster.scheduler.schedule_unique(self._tablet_refresh_time, self._refresh_tablets, connection)
```

Now in pool.py / policies.py you can just call `check_tablets_enabled()` and if it returns true - we know that tablets are supported and were successfully fetched so you can use `_tablets`.

Reply:
Done